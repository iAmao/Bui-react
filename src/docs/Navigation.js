import React from 'react';
import PropTypes from 'prop-types';

import Link from './Link';

const Navigation = ({ components, path: fullPath }) => {
    const path = fullPath.substr(1);

    return (
        <div>
            <div id="navigation-logo">
                <h4>bui React</h4>
            </div>
            <div id="navigation-items">
                <ul className="navigation">
                  <li className={`bui-react-docs-component-name ${path === '' ? 'active': ''}`}>
                    <Link href="/">Getting started</Link>
                  </li>
                  <li className={`bui-react-docs-component-name ${path === 'theming' ? 'active': ''}`}>
                    <Link href="/theming">Theming</Link>
                  </li>
                    {components.map(name => (
                        <li
                          key={name}
                          className={
                            `bui-react-docs-component-name ${path === name.toLowerCase() ? 'active': ''}`
                          }>
                            <Link href={`/${name.toLowerCase()}`}>{name}</Link>
                        </li>
                    ))}
                </ul>
            </div>
        </div>
    )
};

Navigation.propTypes = {
    location: PropTypes.shape({
        path: PropTypes.string
    }),
    components: PropTypes.array.isRequired
};

export default Navigation;
