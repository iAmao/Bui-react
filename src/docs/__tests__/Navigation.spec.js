import React from 'react';
import renderer from 'react-test-renderer';

import { shallow, mount } from 'enzyme';

import Navigation from '../Navigation';
import componentData from '../../../config/jest/componentDataMock';

const DATA = componentData.map((component) => component.name);


describe('DOCS: Navigation', () => {

    describe('Snapshot Test', () => {
        test('renders navigation woth a list of props', () => {
            const tree = renderer.create(
                <Navigation components={DATA} path=""/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });


    describe('Unit Test', () => {
        test('renders a clickable link of components', () => {
            const wrapper = shallow(
                <Navigation components={DATA} path="" />
            );
            expect(wrapper.find('li')).toHaveLength(DATA.length + 2);
        });

        test('sets class of "Getting Started" link to active', () => {
          const wrapper = mount(
            <Navigation components={DATA} path="/" />
          );
          expect(wrapper.find('.active')).toHaveLength(1);
          expect(wrapper.find('.active Link').text()).toEqual('Getting started');
        });

        test('sets the class of "Button" component to active', () => {
          const wrapper = mount(
            <Navigation components={DATA} path="/button" />
          );
          expect(wrapper.find('.active')).toHaveLength(1);
          expect(wrapper.find('.active > Link').text()).toEqual('Button');
        });

    });
});