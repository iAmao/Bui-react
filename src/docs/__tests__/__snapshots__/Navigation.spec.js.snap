// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOCS: Navigation Snapshot Test renders navigation woth a list of props 1`] = `
<div>
  <div
    id="navigation-logo"
  >
    <h4>
      bui React
    </h4>
  </div>
  <div
    id="navigation-items"
  >
    <ul
      className="navigation"
    >
      <li
        className="bui-react-docs-component-name active"
      >
        <a
          onClick={[Function]}
        >
          Getting started
        </a>
      </li>
      <li
        className="bui-react-docs-component-name "
      >
        <a
          onClick={[Function]}
        >
          Theming
        </a>
      </li>
      <li
        className="bui-react-docs-component-name "
      >
        <a
          onClick={[Function]}
        >
          Button
        </a>
      </li>
      <li
        className="bui-react-docs-component-name "
      >
        <a
          onClick={[Function]}
        >
          Icon
        </a>
      </li>
    </ul>
  </div>
</div>
`;
